neo4j-to-neptune/src/test/java/com/amazonaws/services/neptune/metadata/BulkLoadConfigTest.java

```diff
@@ -0,0 +1,354 @@
+            writer.write("iam-role-arn: \"" + TestDataProvider.IAM_ROLE_ARN + "\"\n");
+            writer.write("parallelism: \"" + TestDataProvider.BULK_LOAD_PARALLELISM_LOW + "\"\n");
+            writer.write("monitor: " + TestDataProvider.BULK_LOAD_MONITOR_TRUE+ "\n");
+        }
+
+        BulkLoadConfig config = BulkLoadConfig.fromFile(tempFile);
+
+        // Test all fields are loaded correctly
+        assertEquals(TestDataProvider.BUCKET, config.getBucketName());
+        assertEquals(TestDataProvider.S3_PREFIX, config.getS3Prefix());
+        assertEquals(TestDataProvider.NEPTUNE_ENDPOINT, config.getNeptuneEndpoint());
+        assertEquals(TestDataProvider.IAM_ROLE_ARN, config.getIamRoleArn());
+        assertEquals(TestDataProvider.BULK_LOAD_PARALLELISM_LOW, config.getParallelism());
+        assertTrue(config.isMonitor());
+    }
+
+    @Test
+    public void testDefaultYamlConfiguration() throws IOException {
+        // Test with only required fields
+        File tempFile = File.createTempFile("test-default", ".yaml");
+        tempFile.deleteOnExit();
+
+        try (FileWriter writer = new FileWriter(tempFile)) {
+            writer.write("bucket-name: \"" + TestDataProvider.BUCKET + "\"\n");
+            writer.write("neptune-endpoint: \"" + TestDataProvider.NEPTUNE_ENDPOINT + "\"\n");
+            writer.write("iam-role-arn: \"" + TestDataProvider.IAM_ROLE_ARN + "\"\n");
+        }
+
+        BulkLoadConfig config = BulkLoadConfig.fromFile(tempFile);
+
+        // Test required fields
+        assertEquals(TestDataProvider.BUCKET, config.getBucketName());
+        assertEquals(TestDataProvider.NEPTUNE_ENDPOINT, config.getNeptuneEndpoint());
+        assertEquals(TestDataProvider.IAM_ROLE_ARN, config.getIamRoleArn());
+
+        // Test default values for optional fields
+        assertEquals("", config.getS3Prefix());
+        assertEquals("OVERSUBSCRIBE", config.getParallelism());
+        assertFalse(config.isMonitor()); // boolean default is false
+    }
+
+    @Test
+    public void testEmptyYamlFile() throws IOException {
+        // Test with empty YAML file
+        File tempFile = File.createTempFile("test-empty", ".yaml");
+        tempFile.deleteOnExit();
+
+        try (FileWriter writer = new FileWriter(tempFile)) {
+            // Write empty file
+        }
+
+        BulkLoadConfig config = BulkLoadConfig.fromFile(tempFile);
+
+        // CLI missing parameters validation will kick in all fields should be null values
+        assertNull(config.getBucketName());
+        assertNull(config.getS3Prefix());
+        assertNull(config.getNeptuneEndpoint());
+        assertNull(config.getIamRoleArn());
+        assertNull(config.getParallelism());
+        assertFalse(config.isMonitor()); // boolean default is false
+    }
+
+    @Test
+    public void testNullFile() throws IOException {
+        // Test with null file
+        BulkLoadConfig config = BulkLoadConfig.fromFile(null);
```

Comment:
This has me wondering what happens if the user provides an invalid path for the config file (non-existent/directory instead of file/unreadable file). Does it give a clear error or does it fail silently and set all values to null?